VENDORING.md

```diff
@@ -0,0 +1,232 @@
+that could result from a package change on the upstream.
+
+### Update a vendored package
+
+The following case describes a situation where you add some code that relies
+on another version of a package already vendored. This is a simple case since
+the package is already part of the vendoring. You modify `Gopkg.toml` by
+updating the vendoring version:
+
+```diff
+--- Gopkg.toml.orig     2018-01-23 09:04:54.160760426 -0800
++++ Gopkg.toml.new      2018-01-23 09:05:09.164094911 -0800
+@@ -1,3 +1,3 @@
+ [[constraint]]
+   name = "github.com/foo/bar"
+-  revision = "f5742cb6"
++  revision = "a4be45c7"
+```
+
+Let `dep` update the vendoring by relying on this new revision:
+```bash
+$ dep ensure
+$ dep prune
+```
+
+### Remove a vendored package
+
+The following case describes a situation where you modify the existing
+code, no longer relying on a vendored package. Running the usual commands
+removes the vendored package from `vendor` directory and from
+`Gopkg.lock`:
+```bash
+$ dep ensure
+$ dep prune
+```
+
+If a constraint is set in `Gopkg.toml`, it is your responsibility to remove
+the constraint manually from the file.
+
+### Test a pending pull request
+
+The following case describes a situation where you need to perform testing
+on your code due to a pending pull request from a vendored package.
+The easiest and safest way to do that is to rely on a feature proposed
+by `dep` called `override`. This feature allows you to leave the constraints
+unmodified, but relies on a different version of the vendored package for
+testing purposes.
+
+#### From the main repository
+
+The pending pull request can be submitted to the main repository through a
+development branch `work-dev`. In this case, you will need to modify
+`Gopkg.toml` as follows:
+
+```diff
+--- Gopkg.toml.orig     2018-01-23 09:04:54.160760426 -0800
++++ Gopkg.toml.new      2018-01-23 09:06:49.735633780 -0800
+@@ -1,3 +1,7 @@
+ [[constraint]]
+   name = "github.com/foo/bar"
+   revision = "f5742cb6"
++
++[[override]]
++  name = "github.com/foo/bar"
++  branch = "work-dev"
+```
+
+#### From a forked repository
+
+The pending pull request can be submitted to a forked repository through a
+development branch called `work-dev`. In this case you will need to modify
+`Gopkg.toml` as follows:
+
+```diff
+--- Gopkg.toml.orig     2018-01-23 09:04:54.160760426 -0800
++++ Gopkg.toml.new      2018-01-23 09:09:58.475261707 -0800
+@@ -1,3 +1,8 @@
+ [[constraint]]
+   name = "github.com/foo/bar"
+   revision = "f5742cb6"
++
++[[override]]
++  name = "github.com/foo/bar"
++  branch = "work-dev"
++  source = "github.com/myfork/bar"
+```
+
+The previous case is more common because typically you do not have permissions
+to push directly to the main repository.
+
+## Miscellaneaous
+
+For additional information about vendoring, see
+"[Re-vendor PRs](https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs)".
```

Comment:
Ah - thanks. I was actually thinking of referring to this file from `CONTRIBUTING.md` but good to have it this way round too. Could you just add a ref to the other doc?

Reply:
done